extensions/notification/NotificationConfigManager.js

```diff
@@ -1,283 +1,79 @@
-            } catch (error) {
-                return cb(error);
-            }
-        });
+        } else {
+            this._usesZookeeperBackend = true;
+            this._configManagerBackend = new ZookeeperConfigManager({
+                zkClient,
+                logger,
+            });
+        }
     }
 
     /**
-     * Handler for the change stream "change" event.
-     * Invalidates cached bucket configs based on the change.
-     * @param {ChangeStreamDocument} change Change stream change object
-     * @returns {undefined}
+     * Get bucket notification configuration
+     *
+     * @param {String} bucket - bucket
+     * @param {function} [cb] - callback
+     * @return {Object|undefined} - configuration if available or undefined
      */
-    _handleChangeStreamChangeEvent(change) {
-        // invalidating cached notification configs
-        const cachedConfig = this._cachedConfigs.get(change.documentKey._id);
-        const bucketNotificationConfiguration = change.fullDocument ? change.fullDocument.value.
-            notificationConfiguration : null;
-        switch (change.operationType) {
-            case 'delete':
-                if (cachedConfig) {
-                    this._cachedConfigs.remove(change.documentKey._id);
-                    onConfigManagerCacheUpdate('delete');
-                }
-                break;
-            case 'replace':
-            case 'update':
-                if (cachedConfig) {
-                    // add() replaces the value of an entry if it exists in cache
-                    this._cachedConfigs.add(change.documentKey._id, bucketNotificationConfiguration);
-                    onConfigManagerCacheUpdate('update');
-                }
-                break;
-            default:
-                this._logger.debug('Skipping unsupported change stream event', {
-                    method: 'NotificationConfigManager._handleChangeStreamChange',
-                });
-                break;
+    getConfig(bucket, cb) {
+        const val = this._configManagerBackend.getConfig(bucket);
+        if (!cb) {
+            return val;
+        }
+        if (val instanceof Promise) {
+            return val.then(res => cb(null, res)).catch(err => cb(err));
         }
-        this._logger.debug('Change stream event processed', {
-            method: 'NotificationConfigManager._handleChangeStreamChange',
-        });
+        return cb(null, val);
     }
 
     /**
-     * Initializes a change stream on the metastore collection
-     * Only document delete and update/replace operations are
-     * taken into consideration to invalidate cache.
-     * Newly created buckets (insert operations) are not cached
-     * as queue populator instances read from different kafka
-     * partitions and so don't need the configs for all buckets
-     * @returns {undefined}
+     * Add/update bucket notification configuration.
+     * Only supported in the Zookeeper backend.
```

Comment:
not really true : we have implemented an abstraction, the method is supported on every backend... (even if some backends actually have an empty implementation)